articles/step-by-step-on-how-to-user-nodejs-repl-tutorial/index.md

```diff
@@ -0,0 +1,211 @@
+It captures the user's JavaScript code inputs, interprets and evaluates the result of this code, displays the result to the screen, and repeats the process till the user quits the shell.  
+It's also important to note that this tool does not require file creation to write codes. It comes with ready to use Node.js development environment.
+
+In this tutorial, we learn the basics of Node.js REPL, how we can use this amazing tool to run scripts without creating `.js` files.
+
+### Table of contents
+1. Introduction to REPL
+2. Basic Arithmetic operations using REPL
+3. Creating methods and variables using REPL
+
+### Prerequisites
+
+This tutorial requires:
+1. Node.js Installed in your development local machine.
+2. Basic knowledge of JavaScript and Node.js.
+
+### Objectives
+At the end of this tutorial, you should be able to use REPL tool to run your Node.js scripts.
+
+### Getting started with REPL
+
+Enter the following in the terminal:
+
+```bash
+$ node
+
+```
+Output:
+
+```bash
+> 
+```
+Entering the command `node` in the terminal starts the REPL command-line tool indicated by the `>` symbol.  
+This symbol acts as an indicator that JavaScript is ready to read and evaluate your code.  
+
+We can test the functionality by printing 'hello world' using REPL as shown below:
+
+```bash
+$ node
+> let hello = 'hello world';
+undefined
+> hello
+'hello world'  // output
+> 
+
+```
+In the script, we declare a variable `hello` and assign the text 'hello world'.  
+Next, we call this variable and press enter, to print the 'hello world' string on the screen. That's it.  
+You alos notice on pressing ENTER, `undefined` is printed. This is because `hello` variable doesn't return a value.
+
+Now how do we exit REPL?
+To exit this tool, press `Ctrl+C` on your keyboard.  
+
+Output:
+```bash
+> let hello = 'hello world';
+undefined
+> hello
+'hello world'
+> 
+(To exit, press Ctrl+C again or Ctrl+D or type .exit)
+> 
+
+```
+As shown in the output, you can as well use `ctrl+D` or type `.exit`.  
+
+### Executing code using REPL.
+
+As discussed previously, this tool simplifies work by providing a quick way to test Node.js code without creating files.  
+> It's also important to note that any valid JavaScript code can be executed using REPL.
+
+#### Performing Arithmetical operations in REPL
+In the previous example, we printed 'hello world' string on the screen, in this section let's dive in and look at arithmetics:  
+
+In your terminal, start REPL: 
+
+```bash
+$ node
+> 
+
+```
+Let's perform basic addition, subtraction,modulus, division and multipliation respectively:
+Output:  
+
+```bash
+> 2+2
+4
+> 5-2
+3
+> 10%3
+1
+> 20/2
+10
+> 10*2
+20
+> 
+
+```
+#### Performing operations using Node's Math library
+Math library is an object with defined methods and properties. We can therefore use these features to perform some arithmetics.  
+
+Example:
+
+```bash
+> function getRandomNumber(maximum){
```

Comment:
Please split this into two parts as @srishilesh had suggested.

https://github.com/section-engineering-education/engineering-education/pull/2059#discussion_r626542548